bh1750 i2c light sensor driver + pwm cleanup

andree182 opened the pull request:
Hi, this is a "preliminary" driver pull request - so that others know the driver already exist and they don't have to rewrite it (though it's trivial).

Since I did change extras/i2c lib a bit, and there's pull request #321 in the queue for similar functionality (but cleaner), I'll rebase this once #321 is merged - basically the commit 0b1197b can be dropped.

UncleRus commented on extras/bh1750/bh1750.h:
```diff
@@ -0,0 +1,65 @@
+
+// Start measurement at 1lx resolution. Measurement time is approx 120ms.
+// Device is automatically set to Power Down after measurement.
+#define BH1750_ONE_TIME_LOW_RES_MODE  0x23
```
Better to use enums instead of all this defines

UncleRus replied:
Better to declare enums instead of all this macros.
Something like
```C
typedef enum
{
     BH1750_MODE_CONTINUOUS_HIGH_RES = 0x10,
     ...
} bh1750_mode_t;
```

UncleRus commented:
It would be great if you add simple example.

Thank you for your contribution!

panoti commented:
Why isn't this pull-request not merged?

andree182 commented:
Waiting for #321 first (to rebase this...), then I'll also add an example..

andree182 commented:
Hi guys. I reworked the patches to match new i2c, added example, and added a small cleanup of PWM. I didn't do the enum thing, since the flags can now be combined (for which enums are a bit dangerous - although here only 1 byte stuff is moved around).
